docs/Course Wiki/CS Course Offerings/CS374A.md

```diff
@@ -0,0 +1,95 @@
+
+### Discussion
+
+CS374A has discussion sections on Wednesday and Friday, where students work with each other and a TA to solve practice questions. While optional, these discussions provide a way to get comfortable with content before homework and exams.
+
+### Homework and Guided Problem Sets
+
+CS374A has a homework assignment during each non-exam week, which students may complete in groups of up to three people. Assignments consist of 2-3 problems, and can be difficult, so it is beneficial to start early.
+
+During each non-exam week, students are assigned a Guided Problem Set (GPS) on PrairieLearn to complete individually. GPSes offer instant feedback and focus on building intution necessary to complete homework assignments, so it's recommended to complete the GPS before the homework assignment.
+
+### Exams and Grading
+
+CS374A has two midterms and one final, and typically does not curve exam scores. Exam scores make up roughly 70% of a student's grade, and the remainder comes from homework scores. The lowest 4-6 homework problem scores over the semester are not included in grade calculation, and one GPS is considered equal to one homework problem.
+
+## Instructors
+
+The instructors for CS374A rotate between CS theory staff:
+
+- Sariel Har-Peled: considered to be the most difficult CS374A professor, but usually writes interesting problems and can be a fun lecturer
+- Jeff Erickson: author of the CS374 textbook and an engaging lecturer, some students wait for him to be teaching before taking the course
+- Chandra Chekuri: an engaging lecturer that cares about students and writes reasonable exams
+- Ruta Mehta and Timothy Chan: usually co-teaching with Professor Mehta lecturing, strong lectures but homework and exam questions may be difficult
+
+## Course Tips
+
+It is important to work on homework assignments as a group, with others you work well with. Some groups prefer to solve problems together, while others prefer to break up the problems between members. Whichever approach you take, make sure to understand the solutions to every homework problem, because they may contain techniques that will be useful on exams.
+
+You can save several hours on homework assignments by attending discussion sections for two hours per week. Since discussion attendance is optional, you can go to a different discussion section you are registered if there's space. Discussion sections are also a good place to meet potential groupmates for homework assignments.
+
+It is important to show your work on all assignments, and if you handwrite, write legibly. Graders are generous with partial credit.
+
+## Life After
+
+CS374 is a prerequisite for [CS421 (Programming Languages)](./CS421.md), a required course for Computer Science majors. Students who enjoyed the middle third of the course will enjoy [CS473 (Algorithms)](./CS473.md). Students who enjoyed the first and last thirds of the course will enjoy [CS475 (Formal Models of Computation)](./CS475.md).
```

Comment:
Maybe also add something like "This course really helps conceptualize, break down, and solve problems which is an important skill for engineers to have".